### PR TITLE
[Merged by Bors] - feat: strengthen `instTotallyDisconnectedSpaceOfIsUltrametricDist` to `TotallySeparatedSpace`

### PR DESCRIPTION
**Pull request by alreadydone**
Rename the file it's in to `TotallySeparated`. Also add two lemmas `isClopen_of_disjoint_cover_open` and `totallySeparatedSpace_iff_exists_isClopen`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Connected/TotallyDisconnected.lean [hunk 1]
```diff
@@ -210,15 +210,19 @@ instance (priority := 100) TotallySeparatedSpace.of_discrete (α : Type*) [Topol
   ⟨fun _ _ b _ h => ⟨{b}ᶜ, {b}, isOpen_discrete _, isOpen_discrete _, h, rfl,
     (compl_union_self _).symm.subset, disjoint_compl_left⟩⟩
 
+theorem totallySeparatedSpace_iff_exists_isClopen {α : Type*} [TopologicalSpace α] :
+    TotallySeparatedSpace α ↔ ∀ x y : α, x ≠ y → ∃ U : Set α, IsClopen U ∧ x ∈ U ∧ y ∈ Uᶜ := by
+  simp_rw [totallySeparatedSpace_iff, IsTotallySeparated, Set.Pairwise, Set.mem_univ, true_implies]
+  refine forall₃_congr fun x y _ ↦ ?_
+  refine ⟨fun ⟨U, V, hU, hV, Ux, Vy, f, disj⟩ ↦ ?_, fun ⟨U, hU, Ux, Ucy⟩ ↦ ?_⟩
+  · refine ⟨U, isClopen_of_disjoint_cover_clopen f hU hV disj,
```
Comment (loefflerd) [inline Mathlib/Topology/Connected/TotallyDisconnected.lean:218]: terminal "refine" could be "exact"?

File: Mathlib/Topology/Clopen.lean [hunk 2]
```diff
@@ -102,6 +102,10 @@ theorem isClopen_inter_of_disjoint_cover_clopen {s a b : Set X} (h : IsClopen s)
   rintro x ⟨hx₁, hx₂⟩
   exact ⟨hx₁, by simpa [not_mem_of_mem_compl hx₂] using cover hx₁⟩
 
+theorem isClopen_of_disjoint_cover_clopen {a b : Set X} (cover : univ ⊆ a ∪ b)
```
Comment (loefflerd) [inline Mathlib/Topology/Clopen.lean:105]: The name seems a bit strange; did you perhaps mean "... of disjoint_cover_open"?

File: Mathlib/Topology/Connected/TotallyDisconnected.lean [hunk 3]
```diff
@@ -210,15 +210,19 @@ instance (priority := 100) TotallySeparatedSpace.of_discrete (α : Type*) [Topol
   ⟨fun _ _ b _ h => ⟨{b}ᶜ, {b}, isOpen_discrete _, isOpen_discrete _, h, rfl,
     (compl_union_self _).symm.subset, disjoint_compl_left⟩⟩
 
+theorem totallySeparatedSpace_iff_exists_isClopen {α : Type*} [TopologicalSpace α] :
+    TotallySeparatedSpace α ↔ ∀ x y : α, x ≠ y → ∃ U : Set α, IsClopen U ∧ x ∈ U ∧ y ∈ Uᶜ := by
+  simp_rw [totallySeparatedSpace_iff, IsTotallySeparated, Set.Pairwise, Set.mem_univ, true_implies]
```
Comment (loefflerd) [inline Mathlib/Topology/Connected/TotallyDisconnected.lean:215]:
```suggestion
  simp only [totallySeparatedSpace_iff, IsTotallySeparated, Set.Pairwise, mem_univ, true_implies]
```

**Review by loefflerd — COMMENTED** [review]
Looks good, thanks for fixing this oversight! Just a few minor comments.

File: Mathlib/Topology/Connected/TotallyDisconnected.lean (hunk 3 above)
Comment (alreadydone) [reply Mathlib/Topology/Connected/TotallyDisconnected.lean]: changed

File: Mathlib/Topology/Connected/TotallyDisconnected.lean (hunk 1 above)
Comment (alreadydone) [reply Mathlib/Topology/Connected/TotallyDisconnected.lean]: good catch!

File: Mathlib/Topology/Clopen.lean (hunk 2 above)
Comment (alreadydone) [reply Mathlib/Topology/Clopen.lean]: Hmm, I copied the name of the previous theorem, which involves an additional clopen set `s` ... changed now.

**Comment by loefflerd** [comment]
Looks good now, but please update the PR header to reflect the renaming (since that will become the commit message).

**Comment by alreadydone** [comment]
Done, thanks for the review!

**Comment by loefflerd** [comment]
Looks good now, thanks!

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**